src/sgame/components/HealthComponent.cpp

```diff
@@ -111,19 +108,42 @@ Util::optional<Vec3> direction, int flags, meansOfDeath_t meansOfDeath) {
 		if (entity.oldEnt->s.eType == entityType_t::ET_BUILDABLE && source->client &&
 		    meansOfDeath != MOD_DECONSTRUCT && meansOfDeath != MOD_SUICIDE &&
 		    meansOfDeath != MOD_REPLACE) {
-			if (G_OnSameTeam(entity.oldEnt, source) && !g_friendlyBuildableFire.integer) {
+			if (G_OnSameTeam(entity.oldEnt, source) && !g_friendlyBuildableFire.Get()) {
 				return;
 			}
 		}
 	}
 
 	float take = amount;
 
+	// reduce team damage according to the team, ignoring buildings.
+	// Note that team damage is reduced only for the *other* members of the
+	// team, otherwise you could simply, say, sit on a grenade and lure
+	// aliens to come to you.
+	if (entity.oldEnt != source && G_OnSameTeam(entity.oldEnt, source)
```

Comment:
This stuff should go inside `if (!(flags & DAMAGE_NO_PROTECTION))`

Reply:
To raise this back from the dead, why should it be moved inside? The only documentation about it I've found is this:
```C
#define DAMAGE_NO_PROTECTION 0x00000004 /**< Game settings don't prevent damage. */
```

Also note these two:
```C
#define DAMAGE_PURE          0x00000001 /**< Amount won't be modified. */
#define DAMAGE_NO_LOCDAMAGE  0x00000008 /**< Don't apply locational modifier. */
```

Do you mean I should move it inside the `if (!(flags & DAMAGE_PURE))` instead?

To go back on `DAMAGE_NO_PROTECTION`, I feel like its both unused, and broken: the only user is the overmind, and it suggests that is supposed to deal as much damage to an armored opponent as it does to an unarmored one, but this is not what is happening: armor protects anyway. Maybe if you can explain the correct behavior or some documentation about the intended usage, I can correct it or document it.

Reply:
To raise this back from the dead, why should it be moved inside? The only documentation about it I've found is this:
```C
#define DAMAGE_NO_PROTECTION 0x00000004 /**< Game settings don't prevent damage. */
```

Also note these two:
```C
#define DAMAGE_PURE          0x00000001 /**< Amount won't be modified. */
#define DAMAGE_NO_LOCDAMAGE  0x00000008 /**< Don't apply locational modifier. */
```

Do you mean I should move it inside the `if (!(flags & DAMAGE_PURE))` instead?

To go back on `DAMAGE_NO_PROTECTION`, I feel like its both unused, and broken: the only user is the overmind, and it suggests that is supposed to deal as much damage to an armored opponent as it does to an unarmored one, but this is not what is happening: armor protects anyway. Maybe if you can explain the correct behavior or some documentation about the intended usage, I can correct it or document it.

Reply:
DAMAGE_NO_PROTECTION means the damage must not be reduced by friendly fire cvars. It is used by other things besides the overmind attack. For example, the patch in its current form causes telefragging not to work - instead of killing the target it only removes 50% of the health.